### PR TITLE
Add options to registration (call: false)

### PR DESCRIPTION
**Pull request by AMHOL**
Addressing #2

### CODE REVIEW

**Comment by AMHOL** [comment]
@solnic WDYT?

File: lib/roda/plugins/container.rb [hunk 1]
```diff
@@ -48,15 +62,26 @@ def inherited(subclass)
...
+          content = container.fetch(key) do
             fail ::Roda::ContainerError, "Nothing registered with the name #{key}"
           end
-          instance.respond_to?(:call) ? instance.call : instance
+
+          if content.call?
+            content.item.call
+          else
+            content.item
+          end
```
Comment (solnic) [inline lib/roda/plugins/container.rb]: I'd just have `content.call` here which encapsulates that logic

**Comment by solnic** [comment]
Makes sense :) I left a small nitty-picky comment ;)

File: lib/roda/plugins/container.rb (hunk 1 above)
Comment (AMHOL) [inline lib/roda/plugins/container.rb]: Makes sense, thanks :smile:

**Pull request merged**